### PR TITLE
Store keybindings properly in keymaps.json and improve conflict checking

### PR DESCRIPTION
**Pull request by slhultgren**
Fixes issue #9087
Fixes issue #9094

#### What it does

* Make sure that we only store API relevant information in the keymaps.json to avoid issues when re-creating the bindings from the keymaps.json #9087 #9094
* Make sure disablements for keymappings are only stored once in keymaps.json. #9094
* Only check conflicting keybindings against usable keybindings. #9094

Contributed by STMicroelectronics

#### How to test

Change some shortcuts (at least 2), observe that no "resolved" properties are stored in the keymaps.
Also, follow steps in #9094.

For example, without this patch keymaps.json might contain:
```
  {
    "command": "textEditor.commands.go.lastEdit",
    "keybinding": "ctrl+alt+left",
    "resolved": [
      {
        "key": {
          "code": "ArrowLeft",
          "keyCode": 37,
          "easyString": "left"
        },
        "ctrl": true,
        "shift": false,
        "alt": true,
        "meta": false
      }
    ],
    "scope": 1
  },
```

With this patch, keymaps.json will contain instead:
```
  {
    "command": "textEditor.commands.go.lastEdit",
    "keybinding": "ctrl+alt+left"
  },
```

#### Review checklist

- [x] as an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

File: packages/core/src/common/keybinding.ts [hunk 1]
```diff
@@ -51,21 +51,49 @@ export interface Keybinding {
 export namespace Keybinding {
 
     /**
-     * Returns with the string representation of the binding.
-     * Any additional properties which are not described on
-     * the `Keybinding` API will be ignored.
+     * Compares two keybingings for equality.
```
Comment (kittaakos) [inline packages/core/src/common/keybinding.ts:54]: keybingings -> keybindings

**Review by kittaakos — APPROVED** [review]
I have verified it; it works as expected. Thank you so much for your help, @slhultgren 👍

**Comment by slhultgren** [comment]
Thanks for the review, I amended the commit and fixed the typo :)

**Comment by kittaakos** [comment]
I consider the `root INFO   Navigator root INFO     1) "before each" hook for "copy file"` test failure unrelated to this PR and merge the changes. Thank you for the fixes, @slhultgren 👍

**Pull request merged**